pkg/vault/test/foundry/fuzz/Swap.medusa.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { IBasePool } from "@balancer-labs/v3-interfaces/contracts/vault/IBasePool.sol";
+import { Rounding } from "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+import { FixedPoint } from "@balancer-labs/v3-solidity-utils/contracts/math/FixedPoint.sol";
+import { InputHelpers } from "@balancer-labs/v3-solidity-utils/contracts/helpers/InputHelpers.sol";
+
+import { BasePoolMath } from "../../../contracts/BasePoolMath.sol";
+import { BalancerPoolToken } from "../../../contracts/BalancerPoolToken.sol";
+
+import "../utils/BaseMedusaTest.sol";
+
+contract SwapMedusaTest is BaseMedusaTest {
+    uint256 internal constant MIN_SWAP_AMOUNT = 1e6;
+
+    int256 internal prevInvariant;
+
+    constructor() BaseMedusaTest() {
+        prevInvariant = computeInvariant();
+        vault.manuallySetSwapFee(address(pool), 0);
+    }
+
+    function optimize_currentInvariant() public view returns (int256) {
+        return -int256(computeInvariant());
+    }
+
+    function property_currentInvariant() public returns (bool) {
+        int256 currentInvariant = computeInvariant();
+        return currentInvariant >= prevInvariant;
+    }
+
+    function computeSwapExactIn(uint256 tokenIndexIn, uint256 tokenIndexOut, uint256 exactAmountIn) public {
```

Comment:
Will we implement the exact out too? Maybe we could add them together in this file, exact in and exact out.